### ISSUE TITLE
[dart2wasm] Incorrect runtime type check result

### ISSUE THREAD

**Issue by osa1** [dart-lang/sdk#51187]
```dart
abstract class N {}

class Z implements N {
  Z();
}

class S<K> implements N {
  N before;
  S(this.before);
}

void main() {
  N n = S<Z>(Z());
  print(n.runtimeType); // S<Z>
  print(n is S<Z>); // true
  print(n is S<S<dynamic>>); // true, should be false
}
```

**Comment by osa1** [comment]
Fixed in https://dart-review.googlesource.com/c/sdk/+/280163.

**Comment by osa1** [comment]
We decided to hold off merging the fix until #51200 is fixed as we don't to break demo apps.
